cdap-proto/src/main/java/co/cask/cdap/proto/metadata/MetadataSearchResultRecordV2.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package co.cask.cdap.proto.metadata;
+
+import co.cask.cdap.api.annotation.Beta;
+import co.cask.cdap.api.metadata.Metadata;
+import co.cask.cdap.api.metadata.MetadataEntity;
+import co.cask.cdap.api.metadata.MetadataScope;
+import co.cask.cdap.proto.id.EntityId;
+import co.cask.cdap.proto.id.NamespacedEntityId;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Represent the Metadata search result record.
+ */
+@Beta
+public class MetadataSearchResultRecordV2 {
+  private final MetadataEntity metadataEntity;
+  private final Map<MetadataScope, Metadata> metadata;
+
+  public MetadataSearchResultRecordV2(NamespacedEntityId entityId) {
+    this(entityId.toMetadataEntity());
+  }
+
+  public MetadataSearchResultRecordV2(MetadataEntity metadataEntity) {
+    this(metadataEntity, Collections.emptyMap());
+  }
+
+  public MetadataSearchResultRecordV2(NamespacedEntityId entityId, Map<MetadataScope, Metadata> metadata) {
+    this(entityId.toMetadataEntity(), metadata);
+  }
+
+  public MetadataSearchResultRecordV2(MetadataEntity metadataEntity, Map<MetadataScope, Metadata> metadata) {
+    this.metadataEntity = metadataEntity;
+    this.metadata = metadata;
+  }
+
+  public NamespacedEntityId getEntityId() {
+    return EntityId.fromMetadataEntity(metadataEntity);
```

Comment:
here it is confusing because this may return the parent of that entity, not the entity itself

Reply:
I added a a comment in the code and clarified the behavior of fromMetadataEntity it does not return parent EntityId.